simulation/amaru-sim/src/simulator/simulate.rs

```diff
@@ -69,7 +72,25 @@ pub struct NodeHandle {
     close: Box<dyn FnMut()>,
 }
 
-#[allow(dead_code)]
+#[allow(unused)]
+pub fn pure_stage_node_handle(
+    mut rx: Receiver<Envelope<EchoMessage>>,
+    stage: StageRef<Envelope<EchoMessage>, (u64, StageRef<Envelope<EchoMessage>, Void>)>,
+    mut running: SimulationRunning,
+) -> anyhow::Result<NodeHandle> {
+    let handle = Box::new(move |msg: Envelope<EchoMessage>| {
+        println!("handle called on message:\n  {:?}", msg);
```

Comment:
I would rather use `tracing` here, it's quite useful to have proper logs even for simulation tests I think?